### PR TITLE
[vcpkg baseline][osgearth/gdal] Re-fix dependency gdal

### PR DESCRIPTION
**Pull request by JackBoosY**
The gdal requires freexl on non-Windows:
```
/mnt/vcpkg-ci/installed/x64-linux/debug/lib/libgdal.a(ogrxlsdatasource.o): In function `OGRXLSDataSource::~OGRXLSDataSource()':
/mnt/vcpkg-ci/buildtrees/gdal/x64-linux-dbg/ogr/ogrsf_frmts/xls/ogrxlsdatasource.cpp:66: undefined reference to `freexl_close'
/mnt/vcpkg-ci/installed/x64-linux/debug/lib/libgdal.a(ogrxlsdatasource.o): In function `OGRXLSDataSource::Open(char const*, int)':
/mnt/vcpkg-ci/buildtrees/gdal/x64-linux-dbg/ogr/ogrsf_frmts/xls/ogrxlsdatasource.cpp:147: undefined reference to `freexl_get_info'
/mnt/vcpkg-ci/buildtrees/gdal/x64-linux-dbg/ogr/ogrsf_frmts/xls/ogrxlsdatasource.cpp:152: undefined reference to `freexl_select_active_worksheet'
/mnt/vcpkg-ci/buildtrees/gdal/x64-linux-dbg/ogr/ogrsf_frmts/xls/ogrxlsdatasource.cpp:155: undefined reference to `freexl_get_worksheet_name'
/mnt/vcpkg-ci/buildtrees/gdal/x64-linux-dbg/ogr/ogrsf_frmts/xls/ogrxlsdatasource.cpp:160: undefined reference to `freexl_worksheet_dimensions'
/mnt/vcpkg-ci/buildtrees/gdal/x64-linux-dbg/ogr/ogrsf_frmts/xls/ogrxlsdatasource.cpp:171: undefined reference to `freexl_close'
/mnt/vcpkg-ci/installed/x64-linux/debug/lib/libgdal.a(ogrxlsdatasource.o): In function `OGRXLSDataSource::GetXLSHandle()':
/mnt/vcpkg-ci/buildtrees/gdal/x64-linux-dbg/ogr/ogrsf_frmts/xls/ogrxlsdatasource.cpp:186: undefined reference to `freexl_open'
/mnt/vcpkg-ci/installed/x64-linux/debug/lib/libgdal.a(ogrxlslayer.o): In function `OGRXLSLayer::DetectHeaderLine(void const*)':
/mnt/vcpkg-ci/buildtrees/gdal/x64-linux-dbg/ogr/ogrsf_frmts/xls/ogrxlslayer.cpp:95: undefined reference to `freexl_get_cell_value'
/mnt/vcpkg-ci/buildtrees/gdal/x64-linux-dbg/ogr/ogrsf_frmts/xls/ogrxlslayer.cpp:105: undefined reference to `freexl_get_cell_value'
/mnt/vcpkg-ci/installed/x64-linux/debug/lib/libgdal.a(ogrxlslayer.o): In function `OGRXLSLayer::DetectColumnTypes(void const*, int*)':
/mnt/vcpkg-ci/buildtrees/gdal/x64-linux-dbg/ogr/ogrsf_frmts/xls/ogrxlslayer.cpp:139: undefined reference to `freexl_get_cell_value'
/mnt/vcpkg-ci/installed/x64-linux/debug/lib/libgdal.a(ogrxlslayer.o): In function `OGRXLSLayer::GetLayerDefn()':
/mnt/vcpkg-ci/buildtrees/gdal/x64-linux-dbg/ogr/ogrsf_frmts/xls/ogrxlslayer.cpp:209: undefined reference to `freexl_select_active_worksheet'
/mnt/vcpkg-ci/buildtrees/gdal/x64-linux-dbg/ogr/ogrsf_frmts/xls/ogrxlslayer.cpp:236: undefined reference to `freexl_get_cell_value'
/mnt/vcpkg-ci/installed/x64-linux/debug/lib/libgdal.a(ogrxlslayer.o): In function `OGRXLSLayer::GetNextRawFeature()':
/mnt/vcpkg-ci/buildtrees/gdal/x64-linux-dbg/ogr/ogrsf_frmts/xls/ogrxlslayer.cpp:317: undefined reference to `freexl_select_active_worksheet'
/mnt/vcpkg-ci/buildtrees/gdal/x64-linux-dbg/ogr/ogrsf_frmts/xls/ogrxlslayer.cpp:324: undefined reference to `freexl_get_cell_value'
```

Fix this.

Since #16088 or #15808 doesn't merged, fix this issue in osgearth temporary.

### CODE REVIEW

**Comment by JackBoosY** [comment]
Done.

**Comment by JackBoosY** [comment]
@vicroms Ping for merge this PR.

File: ports/gdal/portfile.cmake [hunk 1]
```diff
@@ -185,7 +185,7 @@ else()
     endif()
     
     set(CONF_OPTS --enable-shared=${BUILD_DYNAMIC} --enable-static=${BUILD_STATIC})
-    list(APPEND CONF_OPTS --with-proj=${CURRENT_INSTALLED_DIR} --with-libjson-c=${CURRENT_INSTALLED_DIR})
+    list(APPEND CONF_OPTS --with-proj=${CURRENT_INSTALLED_DIR} --with-libjson-c=${CURRENT_INSTALLED_DIR} --with-freexl=${CURRENT_INSTALLED_DIR})
```
Comment (ras0219-msft) [inline ports/gdal/portfile.cmake:188]:
```suggestion
    list(APPEND CONF_OPTS --with-proj=${CURRENT_INSTALLED_DIR} --with-libjson-c=${CURRENT_INSTALLED_DIR} --without-freexl)
```

File: ports/gdal/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  "description": "The Geographic Data Abstraction Library for reading and writing geospatial raster and vector data.",
+  "homepage": "hhttps://gdal.org/",
+  "dependencies": [
+    "cfitsio",
+    "curl",
+    "expat",
+    {
+      "name": "freexl",
+      "platform": "!windows"
+    },
```
Comment (ras0219-msft) [inline ports/gdal/vcpkg.json:14]:
```suggestion
```

**Review by ras0219-msft — CHANGES_REQUESTED** [review]
Since this is a stop-gap, we should approach this with the minimal change and simply disable freexl as a dependency. This will avoid needing to change osgearth as well.

**Pull request merged**